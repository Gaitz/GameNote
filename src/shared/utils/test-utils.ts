import { render } from "@testing-library/react"
import store from "game-note/shared/store"
import theme from "game-note/shared/styles/theme"

// const AllTheProviders = ({ children }) => {
//   return (
//     <ReduxProvider store={store}>
//       <ChakraProvider theme={theme}>{children}</ChakraProvider>
//     </ReduxProvider>
//   )
// }

// const customRender = (ui: ReactElement, options?: object) =>
//   render(ui, { wrapper: AllTheProviders, ...options })

// // re-export everything
// export * from "@testing-library/react"

// // override render method
// export { customRender as render }
