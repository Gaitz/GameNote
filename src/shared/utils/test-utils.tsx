import "@testing-library/jest-dom"
import { render } from "@testing-library/react"
import { Provider as ReduxProvider } from "react-redux"
import { ChakraProvider } from "@chakra-ui/react"
import store from "game-note/shared/store"
import theme from "game-note/shared/styles/theme"
import React, { FC } from "react"
import type { ReactElement } from "react"
import type { RenderOptions } from "@testing-library/react"

/* eslint-disable react/prop-types */
const AllTheProviders: FC = ({ children }) => {
  return (
    <ReduxProvider store={store}>
      <ChakraProvider theme={theme}>{children}</ChakraProvider>
    </ReduxProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
) => render(ui, { wrapper: AllTheProviders, ...options })

export * from "@testing-library/react"
export { default as userEvent } from "@testing-library/user-event"

export { customRender as render }
