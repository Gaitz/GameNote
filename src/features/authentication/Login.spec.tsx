import { render } from "game-note/shared/utils/test-utils"
import { Login } from "./index"

jest.mock("./hooks", () => {
  return {
    __esModule: true,
    useFirebaseAuthLogin: jest.fn(() => ({
      isLoading: false,
      handleGitHubSignIn: jest.fn()
    }))
  }
})

jest.mock("./EmailLogin", () => {
  return {
    __esModule: true,
    default: function mockEmailLogin() {
      return <p>EmailLogin</p>
    }
  }
})

describe("Login component", () => {
  it("Snapshot", () => {
    const { asFragment } = render(<Login />)
    expect(asFragment()).toMatchSnapshot()
  })
})
