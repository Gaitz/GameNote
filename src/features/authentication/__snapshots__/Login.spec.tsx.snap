// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component Snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-151wynj"
  >
    <div
      class="chakra-select__wrapper css-dsdwpa"
    >
      <select
        class="chakra-select css-1xl1c6r"
        data-testid="languageSelector"
      >
        <option
          value=""
        >
          languageSwitch
        </option>
        <option
          value="en"
        >
          English
        </option>
        <option
          value="zh-TW"
        >
          繁體中文
        </option>
      </select>
      <div
        class="chakra-select__icon-wrapper css-zj2dqk"
      >
        <svg
          aria-hidden="true"
          class="chakra-select__icon"
          focusable="false"
          role="presentation"
          style="width: 1em; height: 1em; color: currentColor;"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class="css-g6o626"
  >
    <h1
      class="chakra-heading css-1pwqplt"
    >
      <p
        class="chakra-text css-16l8t46"
      >
        welcome
      </p>
      Game Note
    </h1>
    <div
      class="css-w736gn"
    >
      <button
        class="chakra-button css-h2xeff"
        type="button"
      >
        GitHub login
      </button>
      <button
        class="chakra-button css-h2xeff"
        type="button"
      >
        Email login
      </button>
      <div
        class="chakra-fade"
        style="opacity: 0;"
      >
        <p>
          EmailLogin
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
